### DISCUSSION TITLE
Slow cycle

### DISCUSSION THREAD

**Discussion by marcusfpereira** [nginx-proxy/acme-companion#839] [Q&A]
I have acme-companion to manage around 2500 certificates and 300 containers.

The complete cycle of letsencrypt_service takes more then an hour to complete since it checks all virtual_hosts and certificates.

Some times this time is too long. When I need to add a new virtual_host and the cycle has already started it takes more then 2 hours to get it ready. Eventually the verification fails and the next try will take again another hour.

Is there a way to optimize this cycle?
May be running letsencrypt_service only for the recent changed virtual_hosts or skipping the not changed before the last cycle.

**Comment by buchdag** [comment]
Hi. Unfortunately I don't think that container was ever intended for that kind of heavy production usage. It's basically a bash script, so yes everything get done in sequence without any parallelisation of tasks, and it's not fast.

> May be running letsencrypt_service only for the recent changed virtual_hosts or skipping the not changed before the last cycle.

Right now the container does not persist any state between runs of the service loop so there is currently no way to do this.

I looked into **docker-gen** but it does not have the required sorting capability either.

I'll let you know if I think of something.
